Add Photon Realtime tutorials page

yushimatenjin opened the pull request:
I have written about how to combine the JavaScript SDK with PlayCanvas.

### Links
Photon
https://www.photonengine.com/

### External projects
**1. A simple project for the tutorial .**
~~Project URL: https://playcanvas.com/project/928285/~~
~~Publish URL: https://playcanv.as/e/p/sLGr3Ql2/~~

Project URL: https://playcanvas.com/project/954410/
Publish URL: https://playcanv.as/p/wDUPyoOS/

**2. The full project, including matchmaking.**
Project URL: https://playcanvas.com/project/926999/
Publish URL: https://playcanv.as/e/p/dvJYXs1Y/

### Capture
![localhost_51000_en_tutorials_real-time-multiplayer-photon_ (3)](https://user-images.githubusercontent.com/39250588/174583601-5f89a761-5cef-439e-9fa9-b202b8e272af.png)

yaustar commented on content/en/tutorials/real-time-multiplayer-photon.md:
```diff
@@ -0,0 +1,528 @@
+
+### 1. Using the LoadBalancing class in the SDK
+
+Develop using a class that extends **LoadBalancingClient**.
+Please create **photon-loadbalancing.js**.
```
```suggestion
To get started, we need to create a class that extends **LoadBalancingClient**. Please create a new script asset named **photon-loadbalancing.js** and add the following code:
```

yaustar commented on content/en/tutorials/real-time-multiplayer-photon.md:
```diff
@@ -0,0 +1,528 @@
+
+### 2. Instantiate classes from PlayCanvas to use Photon
+
+Create a **photon-loadbalancing-playcanvas.js** to the project to initialize Photon.
```
```suggestion
Create a script asset named **photon-loadbalancing-playcanvas.js** to the project to initialize Photon.
```

yaustar commented on content/en/tutorials/real-time-multiplayer-photon.md:
```diff
@@ -0,0 +1,528 @@
+
+Create a new **photon-loadbalancing-playcanvas.js** and attach it to the root entity.
+
+![Root Entity - Inspector][16]
```
Missing image from preview

yaustar commented on content/en/tutorials/real-time-multiplayer-photon.md:
```diff
@@ -0,0 +1,528 @@
+
+### 3. Set Script for Root entity
+
+Create a new **photon-loadbalancing-playcanvas.js** and attach it to the root entity.
```
```suggestion
Create a new script asset **photon-loadbalancing-playcanvas.js** and attach it to the Root entity in the Editor.
```

yaustar commented on content/en/tutorials/real-time-multiplayer-photon.md:
```diff
@@ -0,0 +1,528 @@
+
+Enter AppId as a script attribute.
+
+![Script Attributes][17]
```
Missing image from preview

yaustar commented on content/en/tutorials/real-time-multiplayer-photon.md:
```diff
@@ -0,0 +1,528 @@
+
+- **wss** Secure connection via WebSocket.
+- **appId** The value application identifier.
+- **appVersion** Game version. Can be a separate Virtual Application.
```
What's a virtual application?

yaustar commented on content/en/tutorials/real-time-multiplayer-photon.md:
```diff
@@ -0,0 +1,528 @@
+
+- **this.app.fire** [communication][22] between scripts.
+
+### If successful
```
```suggestion
### Done!
```

yaustar commented on content/en/tutorials/real-time-multiplayer-photon.md:
```diff
@@ -0,0 +1,528 @@
+
+### If successful
+
+You can play multiplayer in Photon.
```
```suggestion
You can now play multiplayer in Photon!
```

yaustar commented on content/en/tutorials/real-time-multiplayer-photon.md:
```diff
@@ -0,0 +1,528 @@
+
+![Project][23]
+
+Now you can create a room using Photon and synchronize the location information of the characters.
```
```suggestion
You can create a room using Photon and synchronize the positions of players with each other.
```

yushimatenjin commented:
@yaustar 
Thank you so much for your review !
I appreciate the help with this. 
I’ll revise it.

yushimatenjin commented on content/en/tutorials/real-time-multiplayer-photon.md:
```diff
@@ -0,0 +1,528 @@
+
+Download the SDK from the dashboard.
+
+### 1. Click SDK from the dashboard
+
+![SDK][8]
+
+### 2. Select RealTime JavaScript
+
+![JavaScript SDK][9]
+
+### 3. Click download SDK
+
+![Download SDK][10]
+
+### 4. Unzip the SDK
+
+![Unzip SDK][11]
+
+The SDK will be downloaded in ZIP format, unzip it: `photon-javascript-sdk_v4-1-1-0` → `lib` → **`Photon-Javascript_Emscripten_SDK.min.js`**.
```
Sorry, there was a problem with the way I took the screenshot in Chrome.
I took it again and it worked.

yushimatenjin commented on content/en/tutorials/real-time-multiplayer-photon.md:
```diff
@@ -0,0 +1,528 @@
+- How to use Photon SDK
+- Multiplayer implementation
+
+### Tutorial Project
+
+<iframe loading="lazy" src="https://playcanv.as/e/p/sLGr3Ql2/" title="Real-time Multiplayer with Photon - a simple project"></iframe>
+
+*Arrow keys to move.*
+
+You can find the [tutorial project here][2].
```
Thank you! I will remove this one.

yushimatenjin commented:
@yaustar I am sorry for being late to contact you.

I fixed it.
- Changed from Class to Prototype implementation
- Changed Tutorial
- Retaking screenshots
- Fixed many mistakes

### Capture
-----
![](https://user-images.githubusercontent.com/39250588/177569428-06b14000-dd49-4ce9-ae8f-5e42d08a0c72.png)

yaustar commented:
Looks good on my side.

The game/published build still doesn't create a room for me. Is something extra needed?

https://user-images.githubusercontent.com/16639049/177743157-02efcf1a-ae50-466a-b7df-951cc5be0745.mp4

@willeastcott Can you do a final pass please?

yushimatenjin commented:
https://user-images.githubusercontent.com/39250588/177781124-fa7183ad-efa9-474b-ab86-66250e82c63d.mp4

Maybe the problem is that it is set up for the Japan region.
We have changed the region of the open project from jp to us.

willeastcott reviewed (approved):
Final edit pass done. Looks fantastic! ❤️

yaustar commented:
Cool, it looks like that works!

I did notice that the complete project https://playcanvas.com/project/926999/overview/playcanvas-photon-realtime still has some classes used for the pc scripts etc eg https://playcanvas.com/editor/code/926999?tabs=82205460

Can this be updated to use ES5 style that is done in the tutorial please? After that, we are good to go :)

yushimatenjin commented:
Thank you!
> Can this be updated to use ES5 style that is done in the tutorial please? 
Yes, I will fix it as soon as possible!

yushimatenjin commented:
Fixed !
- classes to prototype

> Cool, it looks like that works!
> 
> I did notice that the complete project https://playcanvas.com/project/926999/overview/playcanvas-photon-realtime still has some classes used for the pc scripts etc eg https://playcanvas.com/editor/code/926999?tabs=82205460
> 
> Can this be updated to use ES5 style that is done in the tutorial please? After that, we are good to go :)

I fixed it!
- all class to prototype

yaustar commented:
Awesome, I get this merged and published :) Many thanks!
